registry/auth/htpasswd/access.go

```diff
@@ -60,7 +53,35 @@ func (ac *accessController) Authorized(ctx context.Context, accessRecords ...aut
 		}
 	}
 
-	if err := ac.AuthenticateUser(username, password); err != nil {
+	// Dynamically parsing the latest account list
+	fstat, err := os.Stat(ac.path)
+	if err != nil {
+		return nil, err
+	}
+
+	lastModified := fstat.ModTime().String()
+	ac.mu.Lock()
+	if ac.modtime != lastModified {
```

Comment:
Can you please also add `|| ac.htpasswd == nil` to make sure we never use a nil `htpasswd` even if the filesystem somehow doesn't return a modification time (perhaps this could happen with a network filesystem that doesn't track modification times).